src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -94,6 +94,10 @@ public class TopicPartitionChannel {
   private final AtomicLong processedOffset =
       new AtomicLong(NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE);
 
+  // The in-memory consumer offset managed by the connector, we need this to tell Kafka which
+  // offset to resend when the channel offset token is NULL
+  private long latestConsumerOffset = NO_OFFSET_TOKEN_REGISTERED_IN_SNOWFLAKE;
```

Comment:
the above `offsetPersistedInSnowflake` and `processedOffset` are AtomicLongs, should this also be an AtomicLong?

Reply:
I believe everything should be long given that there is no concurrent logic?